nats.go

```diff
@@ -3060,22 +3086,32 @@ func (nc *Conn) createNewRequestAndSend(subj string, hdr, data []byte) (chan *Ms
 	// Create new literal Inbox and map to a chan msg.
 	mch := make(chan *Msg, RequestChanLen)
 	respInbox := nc.newRespInbox()
+	customInbox := nc.Opts.CustomInboxPrefix
 	token := respInbox[respInboxPrefixLen:]
 	nc.respMap[token] = mch
 	if nc.respMux == nil {
 		// Create the response subscription we will use for all new style responses.
 		// This will be on an _INBOX with an additional terminal token. The subscription
 		// will be on a wildcard.
-		s, err := nc.subscribeLocked(nc.respSub, _EMPTY_, nc.respHandler, nil, false, nil)
+		var respSub string
+		if customInbox != "" {
+			respSub = strings.Replace(nc.respSub, InboxPrefix, customInbox, 1)
+		} else {
+			respSub = nc.respSub
+		}
+		s, err := nc.subscribeLocked(respSub, _EMPTY_, nc.respHandler, nil, false, nil)
 		if err != nil {
 			nc.mu.Unlock()
 			return nil, token, err
 		}
-		nc.respScanf = strings.Replace(nc.respSub, "*", "%s", -1)
+		nc.respScanf = strings.Replace(respSub, "*", "%s", -1)
 		nc.respMux = s
 	}
 	nc.mu.Unlock()
 
+	if customInbox != "" {
+		respInbox = customInbox + respInbox[inboxPrefixLen:]
```

Comment:
Avoid s + s

Reply:
I updated to use strings.Builder, though `s + s` didn't seem bad on my benchmark for this use case either I think, bench test I used can be found here: https://gist.github.com/wallyqs/37bf7aff6af658127b8d954ff7996b24#file-inbox_concat_test-go

Result:

https://gist.github.com/wallyqs/37bf7aff6af658127b8d954ff7996b24#file-osx-results-txt